frontend/src/metabase-lib/lib/Dimension.ts

```diff
@@ -815,80 +818,127 @@ export class FieldDimension extends Dimension {
     return field;
   }
 
-  field(): Field {
+  _getIdentifierProp() {
+    return this.isIntegerFieldId() ? "id" : "name";
+  }
+
+  _getTrustedFieldCachedOnInstance() {
     if (
       this._fieldInstance &&
       this._fieldInstance._comesFromEndpoint === true
     ) {
       return this._fieldInstance;
     }
+  }
 
-    const question = this.query()?.question();
-    const lookupField = this.isIntegerFieldId() ? "id" : "name";
-    const fieldMetadata = question
-      ? _.findWhere(question.getResultMetadata(), {
-          [lookupField]: this.fieldIdOrName(),
-        })
-      : null;
+  _getFieldMetadataFromSavedDataset() {
+    const identifierProp = this._getIdentifierProp();
+    const questionAssociatedWithDimension = this.query()?.question();
+    if (
+      questionAssociatedWithDimension?.isSaved() &&
+      questionAssociatedWithDimension.isDataset()
+    ) {
+      const question = this.query()?.question();
+      const field = _.findWhere(question.getResultMetadata(), {
```

Comment:
could/should we use `questionAssociatedWithDimension` instead of `question` so we don't have to declare `question` again? As having 2 variables in the same scope that has the same value could be confusing.

Reply:
Good catch -- updated.